### ISSUE TITLE
systems_of_equations_ex8 fail in make check with complex version

### ISSUE THREAD

**Issue by canesin** [libMesh/libmesh#513]
Compiled with:
../configure --prefix=$HOME/libmesh_head  --enable-complex

Raise error:
```bash
Making check in systems_of_equations/systems_of_equations_ex8
/Applications/Xcode.app/Contents/Developer/usr/bin/make  check-am
/Applications/Xcode.app/Contents/Developer/usr/bin/make  example-dbg example-devel example-opt   run.sh
  CXX      example_dbg-systems_of_equations_ex8.o
  CXX      example_dbg-augment_sparsity_on_contact.o
  CXX      example_dbg-linear_elasticity_with_contact.o
../../../../examples/systems_of_equations/systems_of_equations_ex8/linear_elasticity_with_contact.C:222:26: error: assigning to 'double' from incompatible type 'Number'
      (aka 'complex<Real>')
              uvw(index) = value;
                         ^ ~~~~~
1 error generated.
make[4]: *** [example_dbg-linear_elasticity_with_contact.o] Error 1
```

**Comment by jwpeterson** [comment]
On Fri, Mar 27, 2015 at 11:37 PM, Fábio César Canesin <
notifications@github.com> wrote:

> Compiled with:
> ../configure --prefix=$HOME/libmesh_head --enable-complex
>
> Raises:
>
> Making check in systems_of_equations/systems_of_equations_ex8
> /Applications/Xcode.app/Contents/Developer/usr/bin/make  check-am
> /Applications/Xcode.app/Contents/Developer/usr/bin/make  example-dbg example-devel example-opt   run.sh
>   CXX      example_dbg-systems_of_equations_ex8.o
>   CXX      example_dbg-augment_sparsity_on_contact.o
>   CXX      example_dbg-linear_elasticity_with_contact.o
> ../../../../examples/systems_of_equations/systems_of_equations_ex8/linear_elasticity_with_contact.C:222:26: error: assigning to 'double' from incompatible type 'Number'
>       (aka 'complex<Real>')
>               uvw(index) = value;
>                          ^ ~~~~~
> 1 error generated.
> make[4]: *** [example_dbg-linear_elasticity_with_contact.o] Error 1
>
>
Thanks for the report.  This new example was just merged today and doesn't
work in complex mode apparently.

-- 
John

**Comment by jwpeterson** [comment]
Should be fixed in 50d9fff, would appreciate if you can confirm.

**Comment by canesin** [comment]
Can't confirm because it is now giving an error with unique_prt:

```
Fabios-iMac:systems_of_equations_ex8 canesin$ make check
/Applications/Xcode.app/Contents/Developer/usr/bin/make  check-am
/Applications/Xcode.app/Contents/Developer/usr/bin/make  example-dbg example-devel example-opt   run.sh
  CXX      example_dbg-linear_elasticity_with_contact.o
../../../../examples/systems_of_equations/systems_of_equations_ex8/linear_elasticity_with_contact.C:154:36: error: no viable conversion from
      'std::unique_ptr<NumericVector<complex<double> > >' to 'AutoPtr<NumericVector<Number> >'
  AutoPtr< NumericVector<Number> > localized_input_solution =
                                   ^
../../../include/libmesh/auto_ptr.h:185:3: note: candidate constructor not viable: no known conversion from 'std::unique_ptr<NumericVector<complex<double> > >'
      to 'libMesh::AutoPtr<libMesh::NumericVector<std::__1::complex<double> > > &' for 1st argument
  AutoPtr(AutoPtr& a)
  ^
../../../include/libmesh/auto_ptr.h:341:3: note: candidate constructor not viable: no known conversion from 'std::unique_ptr<NumericVector<complex<double> > >'
      to 'AutoPtrRef<element_type>' for 1st argument
  AutoPtr(AutoPtrRef<element_type> ref)
  ^
../../../include/libmesh/auto_ptr.h:200:3: note: candidate template ignored: could not match 'AutoPtr' against 'unique_ptr'
  AutoPtr(AutoPtr<Tp1>& a)
  ^
```

**Comment by dknez** [comment]
This example works fine for me with --enable-complex.

**Comment by canesin** [comment]
@dknez my bot without unique_ptr just finished and it work with the --enable-complex, should I submit a new issue for the unique_ptr setting ??

Also on running ex8 it prints:
Exodus header info retrieved successfully.
Title: 	cubit(F:/Akselos/Codes/ALSTOM/david_thermal_test.exo): 05/27/2013: 11:13:48

Maybe you should clean this (so it don't show the companies names) ??

**Comment by jwpeterson** [comment]
Should be a simple fix for --enable-unique-ptr, I'll take a look when I get back from vacation.

**Comment by dknez** [comment]
Thanks for pointing out the title of the mesh. I'll fix that.

**Comment by jwpeterson** [comment]
systems_of_equations_ex8 `UniquePtr` issues fixed in e00a8a60421ad25e99f35955c36ec260a0643ef9.

**Comment by jwpeterson** [comment]
Mesh title fixed in 3f71a9567110bdb5af3e0879a55d2c897514c093.
